components/script/dom/idbobjectstore.rs

```diff
@@ -385,6 +385,29 @@ impl IDBObjectStore {
         self.key_path.is_some()
     }
 
+    /// Checks if the transation is active, throwing a "TransactionInactiveError" DOMException if not.
+    /// If writing is true, it then checks if the transaction is a read-only transaction, throwing a "ReadOnlyError" DOMException if so.
+    fn check_transaction(&self, writing: bool) -> Fallible<()> {
+        // Let transaction be this object store handle's transaction.
+        let transaction = self
+            .transaction
+            .get()
+            .expect("No transaction in Object Store");
```

Comment:
We should reduce unwrap usage without any warning. This should return error with `error!` log mentioned what's happening.

Reply:
No transaction means an inactive one, to my knowledge, so I just returned that error.